### PR TITLE
do not compile if  not sol files & fix typescript autocompletion

### PR DESCRIPTION
**Pull request by yann300**
- do not compile if not sol files 
 - fix typescript autocompletion

### CODE REVIEW

File: apps/remix-ide/src/app/editor/editor.js [hunk 1]
```diff
@@ -150,6 +151,25 @@ class Editor extends Plugin {
...
+        // so Monaco can provide auto completion
+        let content = await this.call('fileManager', 'readFile', name)
+        const paths = name.split('/')
+        paths.pop()
+        const fromPath = paths.join('/') // get current execution context path
+        for (const match of content.matchAll(/import\s+.*\s+from\s+(?:"(.*?)"|'(.*?)')/g)) {
+          let path = match[2]
+          if (path.startsWith('./') || path.startsWith('../')) path = resolve(fromPath, path)
+          if (path.startsWith('/')) path = path.substring(1)
+          if (!path.endsWith('.ts')) path = path + '.ts'
```
Comment (Aniket-Engg) [inline apps/remix-ide/src/app/editor/editor.js:167]: This creates an issue for npm module import like ethers and web3 and shows errors:
Screenshot 2022-05-31 at 12 39 52 PM

File: libs/remix-ui/solidity-compiler/src/lib/logic/compileTabLogic.ts [hunk 2]
```diff
@@ -131,6 +131,11 @@ export class CompileTabLogic {
 
   runCompiler (externalCompType) {
     try {
+      const currentFile = this.api.currentFile
+      if (currentFile && (currentFile.endsWith('.ts') || currentFile.endsWith('.js') || currentFile.endsWith('.txt') || currentFile.endsWith('.log'))) {
```
Comment (Aniket-Engg) [inline libs/remix-ui/solidity-compiler/src/lib/logic/compileTabLogic.ts:135]: I think it is better to check files available for compilation like .sol, .vy etc as there are a great number of other extensions possible.

I can still compile the .json and .md file

Comment (yann300) [reply libs/remix-ui/solidity-compiler/src/lib/logic/compileTabLogic.ts:135]: yes I know, i'll find an alternative

File: apps/remix-ide/src/app/editor/editor.js (hunk 1 above)
Comment (yann300) [reply apps/remix-ide/src/app/editor/editor.js:167]: done

**Review by ioedeveloper — APPROVED** [review]

**Review by Aniket-Engg — COMMENTED** [review]
Compilation is not triggered for non-sol files that is fine. But autocomplete is still not working for an lib imported in an imported file.

**Review by Aniket-Engg — APPROVED** [review]

**Pull request merged**